File: codes/javascript/operator_ternary.js
console.log('\n// Ternary Operator in JavaScript');

const CORRECT_ANSWER = 'foo';
console.log(`correct answer: "${CORRECT_ANSWER}"`);

let myAnswer;
let isMyAnswerCorrect;

myAnswer = 'hello';
console.log(`my answer: "${myAnswer}"`);
isMyAnswerCorrect = myAnswer === CORRECT_ANSWER ? 'congrats!' : 'wrong answer!';
console.log(`result: ${isMyAnswerCorrect}`);
// result: wrong answer!

myAnswer = 'foo';
console.log(`my answer: "${myAnswer}"`);
isMyAnswerCorrect = myAnswer === CORRECT_ANSWER ? 'congrats!' : 'wrong answer!';
console.log(`result: ${isMyAnswerCorrect}`);
// result: congrats!

File: codes/r/operator_ternary.r
library(jsonlite)

prettyJsonStringify <- function(anything) (if (is.null(anything) == TRUE) "NULL" else if (is.character(anything) == TRUE) paste(sep = "", "\"", anything, "\"") else prettify(toJSON(anything, pretty = TRUE, auto_unbox = TRUE), indent = 4))

cat("\n# Ternary Operator in R\n")

CORRECT_ANSWER <- "foo"
cat(paste(sep = "", "correct answer: ", prettyJsonStringify(CORRECT_ANSWER), "\n"))

myAnswer <- "bar"
cat(paste(sep = "", "my answer: ", myAnswer, "\n"))
cat(paste(sep = "", "is my answer correct: ", (if (myAnswer == CORRECT_ANSWER) "correct!" else "wrong!"), "\n"))
# is my answer correct: wrong!

myAnswer <- "foo"
cat(paste(sep = "", "my answer: ", myAnswer, "\n"))
cat(paste(sep = "", "is my answer correct: ", (if (myAnswer == CORRECT_ANSWER) "correct!" else "wrong!"), "\n"))
# is my answer correct: correct!
